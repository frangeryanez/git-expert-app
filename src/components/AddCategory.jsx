import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = event => {
    event.preventDefault();
    const value = inputValue.trim();
    if (value.length <= 1) return;

    // setCategories(categories => [inputValue, ...categories]);
    onNewCategory(value);
    setInputValue('');
  };

  return (
    <form onSubmit={ onSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
};