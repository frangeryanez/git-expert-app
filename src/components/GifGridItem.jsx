import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img 
        alt={ title } 
        id={id} 
        src={ url } 
      />
      <p>{ title }</p>
    </div>
  );
};

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};