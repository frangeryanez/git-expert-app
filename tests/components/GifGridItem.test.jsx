import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('Test on file <GifGridItem />', () => {
  const id = 'image_01';
  const title = 'Title';
  const url = 'https://api.giphy.com/v1/gifs/';
  
  test('Must match the snapshot', () => {
    const { container } = render(
      <GifGridItem 
        id={ id }
        title={ title }
        url={ url }
      />);
    expect(container).toMatchSnapshot();
  });

  test('Must show the image with the indicated URL and ALT', () => {
    render(<GifGridItem id={ id } title={ title } url={ url }/>);
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Should display the title in the component', () => {
    render(<GifGridItem id={ id } title={ title } url={ url }/>);
    expect(screen.getByText(title)).toBeTruthy();
  });
});