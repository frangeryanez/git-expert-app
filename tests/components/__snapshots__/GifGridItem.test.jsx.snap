// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file <GifGridItem /> Must match the snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <img
      alt="Title"
      id="image_01"
      src="https://api.giphy.com/v1/gifs/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
